### PR TITLE
moved featured button to VueJS

### PR DESCRIPTION
**Pull request by vinyll**
_featured_ button from https://github.com/opendatateam/udata/issues/526 list

### CODE REVIEW

File: js/components/dataset/featured.vue [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+</template>
+
+<script>
+import Auth from 'auth';
+import log from 'logger';
+
+export default {
+    data() {
+        return {
+            featured: false
```
Comment (noirbizarre) [inline js/components/dataset/featured.vue]: I think this should be a property in order to be able to give the initial state

File: udata/templates/dataset/display.html [hunk 2]
```diff
@@ -33,15 +33,9 @@
...
 {% if sysadmin %}
-<div class="btn-group btn-group-xs">
-    <button type="button"
-        class="btn btn-success featured {% if dataset.featured %}active{% endif %}"
-        data-api-url="{{ url_for('api.dataset_featured', dataset=dataset) }}">
-        <span class="glyphicon glyphicon-bullhorn"></span>
-        {{ _('Featured') }}
-    </button>
-</div>
+<featured dataset-id="{{ dataset.id }}"></featured>
```
Comment (noirbizarre) [inline udata/templates/dataset/display.html]: Here, you lost the initial state. It was using the active class before, a very weak mecanism. Now, you can have an explicit state `:featured="(false|true)"`.

File: udata/templates/dataset/display.html [hunk 3]
```diff
@@ -167,14 +161,7 @@ <h4 class="list-group-item-heading">
...
                         {% if sysadmin %}
-                        <div class="btn-group btn-group-sm">
-                            <button type="button"
-                                class="btn btn-success featured {% if dataset.featured %}active{% endif %}"
-                                data-api-url="{{ url_for('api.dataset_featured', dataset=dataset) }}">
-                                <span class="glyphicon glyphicon-bullhorn"></span>
-                                {{ _('Featured') }}
-                            </button>
-                        </div>
+                        <featured dataset-id="{{ dataset.id }}"></featured>
```
Comment (noirbizarre) [inline udata/templates/dataset/display.html]: Same as above

**Comment by davidbgk** [comment]
You need to resolve conflict merges 😉

File: js/front/dataset/index.js [hunk 4]
```diff
@@ -7,6 +7,10 @@ import Auth from 'auth';
 import Vue from 'vue';
 import config from 'config';
 import log from 'logger';
+<<<<<<< HEAD
```
Comment (noirbizarre) [inline js/front/dataset/index.js]: I think the merge failed :/

File: js/components/dataset/featured.vue [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<template>
+<div class="btn-group btn-group-sm">
+    <button type="button"
+        class="btn btn-success featured"
+        :class="{active: featured}"
+        @click="toggleFeatured">
+        <span class="glyphicon glyphicon-bullhorn"></span>
```
Comment (noirbizarre) [inline js/components/dataset/featured.vue]: Might be better with font-awesome icons (ie. `fa fa-bullhorn`) so at some point we can drop bootstrap glyphicons totally and gain some Kb in the build

File: udata/templates/dataset/display.html [hunk 6]
```diff
@@ -33,15 +33,9 @@
...
 {% if sysadmin %}
-<div class="btn-group btn-group-xs">
-    <button type="button"
-        class="btn btn-success featured {% if dataset.featured %}active{% endif %}"
-        data-api-url="{{ url_for('api.dataset_featured', dataset=dataset) }}">
-        <span class="glyphicon glyphicon-bullhorn"></span>
-        {{ _('Featured') }}
-    </button>
-</div>
+<featured dataset-id="{{ dataset.id }}" :featured="{% if dataset.featured %}true{% else %}false{% endif %}"></featured>
```
Comment (noirbizarre) [inline udata/templates/dataset/display.html]: Or even simpler with `:featured="{{ dataset.featured|to_json }}"` (it should work) ;)

File: udata/templates/dataset/display.html [hunk 7]
```diff
@@ -162,14 +156,7 @@ <h4 class="list-group-item-heading">
...
                         {% if sysadmin %}
-                        <div class="btn-group btn-group-sm">
-                            <button type="button"
-                                class="btn btn-success featured {% if dataset.featured %}active{% endif %}"
-                                data-api-url="{{ url_for('api.dataset_featured', dataset=dataset) }}">
-                                <span class="glyphicon glyphicon-bullhorn"></span>
-                                {{ _('Featured') }}
-                            </button>
-                        </div>
+                        <featured dataset-id="{{ dataset.id }}" :featured="{% if dataset.featured %}true{% else %}false{% endif %}"></featured>
```
Comment (noirbizarre) [inline udata/templates/dataset/display.html]: Same, cleaner with `to_json`.

**Pull request merged**